addons/binding/org.openhab.binding.regoheatpump/src/main/java/org/openhab/binding/regoheatpump/internal/rego6xx/RegoRegisterMapper.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.regoheatpump.internal.rego6xx;
+
+import static org.openhab.binding.regoheatpump.RegoHeatPumpBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.types.State;;
+
+/**
+ * The {@link RegoRegisterMapper} is responsible for mapping rego 6xx registers into channels.
+ *
+ * @author Boris Krivonog - Initial contribution
+ */
+public class RegoRegisterMapper {
+    public static interface Channel {
+        public short address();
+
+        public State convert(Short value);
+    }
+
+    private static class Int16Channel implements Channel {
+        private final short address;
+
+        private Int16Channel(short address) {
+            this.address = address;
+        }
+
+        @Override
+        public short address() {
+            return address;
+        }
+
+        @Override
+        public State convert(Short value) {
+            return new DecimalType(value);
+        }
+    }
+
+    private static class TemperatureChannel extends Int16Channel {
+        private TemperatureChannel(short address) {
+            super(address);
+        }
+
+        @Override
+        public State convert(Short value) {
+            return new DecimalType(ValueConverter.toDouble(value));
+        }
+    }
+
+    private final Map<String, Channel> mappings;
+
+    private RegoRegisterMapper(Map<String, Channel> mappings) {
+        this.mappings = mappings;
+    }
+
+    public Channel map(String channelIID) {
+        return mappings.get(channelIID);
+    }
+
+    public static RegoRegisterMapper rego600() {
```

Comment:
Why a method that constructs the content on every call again? Simply define a static map in the class directly.